src/main/groovy/com/deploygate/gradle/plugins/internal/http/LocalServer.java

```diff
@@ -0,0 +1,151 @@
+package com.deploygate.gradle.plugins.internal.http;
+
+import com.deploygate.gradle.plugins.internal.credentials.CliCredentialStore;
+import com.deploygate.gradle.plugins.internal.utils.UrlUtils;
+import com.sun.net.httpserver.HttpServer;
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.inject.Inject;
+import org.gradle.api.GradleException;
+import org.gradle.api.provider.Property;
+import org.gradle.api.services.BuildService;
+import org.gradle.api.services.BuildServiceParameters;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class LocalServer implements BuildService<LocalServer.Params>, AutoCloseable {
```

Comment:
This handles notification key based authentication flow.

Reply:
sounds good.